lib/ssz.ex

```diff
@@ -139,4 +164,349 @@ defmodule Ssz do
 
   @spec decode_u256(binary) :: non_neg_integer
   def decode_u256(num), do: :binary.decode_unsigned(num, :little)
+
+  ##### elixir native ssz
+  @bytes_per_length_offset 4
+  @bytes_per_chunk 32
+  @bits_per_byte 8
+
+  ### ENCODE ######
+  def serialize(%struct{} = map) do
+    schemas = struct.schema()
+
+    serialized =
+      schemas
+      |> Enum.map(fn schema ->
+        key = Enum.at(Map.keys(schema), 0)
+        metadata = Map.get(schema, key)
+        value = Map.get(map, key)
+        {:ok, serialized} = serialize(value, metadata)
+        serialized
+      end)
+      |> :binary.list_to_bin()
+
+    {:ok, serialized}
+  end
+
+  def serialize(elements) when is_list(elements) do
+    serialized =
+      elements
+      |> Enum.map(fn element ->
+        {:ok, serialized} = serialize(element)
+        serialized
+      end)
+      |> :binary.list_to_bin()
+
+    {:ok, serialized}
+  end
+
+  def serialize(value) when is_binary(value), do: {:ok, value}
+
+  def serialize(value, []) when is_binary(value), do: {:ok, value}
+
+  def serialize(value, %{type: :struct}) do
```

Comment:
I think we should use `:container` instead of `:struct`, just to follow the spec names.